control-plane/pkg/contract/subscriptionsapi.go

```diff
@@ -0,0 +1,198 @@
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#all-filter-dialect
+func NewAllFilter(filters []v1.SubscriptionsAPIFilter) *DialectedFilter {
+	all := All{
+		Filters: make([]*DialectedFilter, 0, len(filters)),
+	}
+	for _, f := range filters {
+		all.Filters = append(all.Filters, FromSubscriptionFilter(f))
+	}
+	return &DialectedFilter{
+		Filter: &DialectedFilter_All{
+			All: &all,
+		},
+	}
+}
+
+// NewAnyFilter converts the SubscriptionsAPIFilter into the any dialect of the
+// DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// All filter evaluates to true when all nested filters evaluate to true.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#any-filter-dialect
+func NewAnyFilter(filters []v1.SubscriptionsAPIFilter) *DialectedFilter {
+	any := Any{
+		Filters: make([]*DialectedFilter, 0, len(filters)),
+	}
+	for _, f := range filters {
+		any.Filters = append(any.Filters, FromSubscriptionFilter(f))
+	}
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Any{
+			Any: &any,
+		},
+	}
+}
+
+// NewNotFilter converts the SubscriptionsAPIFilter into the not dialect of the
+// DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// Not filter evaluates to true when the nested filter evaluates to false.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#not-filter-dialect
+func NewNotFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Not{
+			Not: &Not{
+				Filter: FromSubscriptionFilter(f),
+			},
+		},
+	}
+}
+
+// NewCESQLFilter converts the SubscriptionsAPIFilter into the sql dialect of the
+// DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// CESOL filter is a Cloud Events SQL Expression
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/sql.md#any-filter-dialect
+//
+// See "CNCF CloudEvents SQL Expression Language"
+// https://github.com/cloudevents/spec/blob/main/cesql/spec.md
+func NewCESQLFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Cesql{
+			Cesql: &CESQL{
+				Expression: f.SQL,
+			},
+		},
+	}
+}
+
+// FromSubscriptionFilter converts a SubscriptionsAPIFilter to the corresponding
+// DialectedFilter based on the property set in the SubscriptionsAPIFilter. If
+// the SubscriptionsAPIFilter has multiple dialects set, they will be wrapped inside
+// a Filter of the type DialectedFilter_All
+//
+// E.g a SubscriptionsAPIFilter with the Exact field set will be converted to a DialectedFilter
+// which Filter is of type DialectedFilter_Exact.
+func FromSubscriptionFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
```

Comment:
This is the only function that needs to be public, can we reduce the scope for others `New*Filter` functions?

Reply:
Fixed